.github/workflows/build.yml

```diff
@@ -1,35 +1,125 @@
 name: Build/Release
+
 on:
   push:
     tags:
-      - '*'
+      - 'v*'
+  workflow_dispatch:
+
 jobs:
-  release:
+  build:
     runs-on: ${{ matrix.os }}
+    
     strategy:
+      fail-fast: false
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        include:
+          - os: macos-latest
+            platform: mac
+          - os: ubuntu-latest
+            platform: linux
+          - os: windows-latest
+            platform: win
 
     steps:
-      - name: Check out Git repository
+      - name: Checkout repository
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      - name: Install Node.js, NPM and Yarn
+      - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          cache: yarn
-          
-        # Note: These flags needed to speed up build
+          node-version: '18'
+          cache: 'yarn'
+
+      # Cache Electron binaries
+      - name: Cache Electron binaries
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cache/electron
+            ~/.cache/electron-builder
+            node_modules/.cache/electron
+          key: ${{ runner.os }}-electron-cache-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-electron-cache-
+
+      # Install dependencies with proper caching
       - name: Install dependencies
-        run: yarn install --frozen-lockfile --prefer-offline
+        run: yarn install --frozen-lockfile --network-timeout 600000
+        env:
+          CYPRESS_INSTALL_BINARY: 0
+          ELECTRON_CACHE: ~/.cache/electron
+          ELECTRON_BUILDER_CACHE: ~/.cache/electron-builder
+
+      # Build the application
+      - name: Build application
+        run: yarn build:prod
+        env:
+          NODE_ENV: production
+
+      # Build Electron app
+      - name: Build Electron app
+        run: yarn electron-builder --${{ matrix.platform }} --publish=never
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          APPLE_ID: ${{ secrets.APPLE_ID }}
```

Comment:
I want to set up notarization and signing of mac artifacts, what do else do I need to provide? Provide guide or comment.

Reply:
Can we use APPLE_APP_SPECIFIC_PASSWORD instead of APPLE_ID_PASS?